.github/workflows/release.yml

```diff
@@ -0,0 +1,74 @@
+name: Release RubyGems
+
+on:
+  push:
+    branches:
+      - release/*
+
+jobs:
+
+  pre-release-check:
+    name: Perform checks before releasing
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Check commit has been pushed on origin/main
+        run: |
+          git fetch --quiet origin main
+          git merge-base --is-ancestor HEAD origin/main
+
+  tests:
+    needs: pre-release-check
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        ruby: ['2.5', '2.6', '2.7', '3.0']
+
+        include:
+          - os: ubuntu-latest
+            ruby: jruby
+
+          - os: macos-latest
+            ruby: '3.0'
+
+          - os: windows-latest
+            ruby: '3.0'
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Run ruby tests
+        uses: ./.github/actions/test-ruby
+        with:
+          ruby-version: ${{ matrix.ruby }}
+
+  publish-rubygem:
+    name: Publish Ruby Gem
+    needs: tests
+    runs-on: ubuntu-latest
+    environment: Release
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Publish ruby gem
+        uses: cucumber/action-publish-rubygem@v1.0.0
+        with:
+          rubygems_api_key: ${{ secrets.RUBYGEMS_API_KEY }}
+
+  create-github-release:
+    name: Create GitHub Release and Git tag
+    needs: publish-rubygem
```

Comment:
could we just do these two in parallel? so make this one depend on `test` instead?

Reply:
I would prefer to do the github release only once we have successfully published the gem
That way we do not create git tags and github release if an issue occurs while publishing the gem